File: solutions/java/src/loggingframework/logappender/DatabaseAppender.java
package loggingframework.logappender;

import loggingframework.LogMessage;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class DatabaseAppender implements LogAppender {
    private final String jdbcUrl;
    private final String username;
    private final String password;

    public DatabaseAppender(String jdbcUrl, String username, String password) {
        this.jdbcUrl = jdbcUrl;
        this.username = username;
        this.password = password;
    }

    @Override
    public void append(LogMessage logMessage) {
        try (Connection connection = DriverManager.getConnection(jdbcUrl, username, password);
             PreparedStatement statement = connection.prepareStatement("INSERT INTO logs (level, message, timestamp) VALUES (?, ?, ?)")) {
            statement.setString(1, logMessage.getLevel().toString());
            statement.setString(2, logMessage.getMessage());
            statement.setLong(3, logMessage.getTimestamp());
            statement.executeUpdate();
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
}

File: solutions/csharp/loggingframework/DatabaseAppender.cs
using System;
//using System.Data.SqlClient;

namespace LoggingFramework
{
    public class DatabaseAppender : LogAppender
    {
        private readonly string _connectionString;

        public DatabaseAppender(string connectionString)
        {
            _connectionString = connectionString;
        }

        public void Append(LogMessage logMessage)
        {
            // try
            // {
            //     using (var connection = new SqlConnection(_connectionString))
            //     {
            //         connection.Open();
            //         var command = new SqlCommand("INSERT INTO logs (level, message, timestamp) VALUES (@level, @message, @timestamp)", connection);
            //         command.Parameters.AddWithValue("@level", logMessage.Level.ToString());
            //         command.Parameters.AddWithValue("@message", logMessage.Message);
            //         command.Parameters.AddWithValue("@timestamp", logMessage.Timestamp);
            //         command.ExecuteNonQuery();
            //     }
            // }
            // catch (SqlException ex)
            // {
            //     Console.WriteLine("DatabaseAppender error: " + ex.Message);
            // }
        }
    }
}